pkg/reconciler/shared/tektonconfig/upgrade/post_upgrade.go

```diff
@@ -70,3 +80,66 @@ func upgradeStorageVersion(ctx context.Context, logger *zap.SugaredLogger, k8sCl
 
 	return nil
 }
+
+// Remove the all  cluster tasks installer sets
+func removeAllClusterTasksInstallerSets(ctx context.Context, logger *zap.SugaredLogger, k8sClient kubernetes.Interface, operatorClient versioned.Interface, restConfig *rest.Config) error {
+
+	if !v1alpha1.IsOpenShiftPlatform() {
+		return nil
+	}
+
+	tisClient := operatorClient.OperatorV1alpha1().TektonInstallerSets()
+
+	ctLS := v1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: installerSetType(customType, "ClusterTask"),
+		},
+	}
+	ctLabel, err := common.LabelSelector(ctLS)
+	if err != nil {
+		return err
+	}
+	if err := tisClient.DeleteCollection(ctx, v1.DeleteOptions{}, v1.ListOptions{
+		LabelSelector: ctLabel,
+	}); err != nil {
+		logger.Error("Failed to delete the ClusterTask installer set", err)
+		return err
+	}
+
+	cctLS := v1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: installerSetType(customType, "CommunityClusterTask"),
+		},
+	}
+	cctLabel, err := common.LabelSelector(cctLS)
+	if err != nil {
+		return err
+	}
+	if err := tisClient.DeleteCollection(ctx, v1.DeleteOptions{}, v1.ListOptions{
+		LabelSelector: cctLabel,
+	}); err != nil {
+		logger.Error("Failed to delete the CommunityClusterTask installer set", err)
+		return err
+	}
+
+	vctLS := v1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: installerSetType(customType, "VersionedClusterTask"),
+		},
+	}
+	vctLabel, err := common.LabelSelector(vctLS)
+	if err != nil {
+		return err
+	}
```

Comment:
Should we move all the labels to the top in the constant and just use in the deletion call ?

Reply:
we can not create as const for those value and it's not a constant